src/components/interactive-builder/add-question.modal.tsx

```diff
@@ -197,7 +213,8 @@ const AddQuestionModal: React.FC<AddQuestionModalProps> = ({
         type: questionType,
         required: isQuestionRequired,
         id: questionId ?? computedQuestionId,
-        ...(questionType === 'encounterDatetime' && { datePickerFormat: datePickerType }),
+        ...((renderingType === 'date' || renderingType === 'datetime') &&
+          datePickerType && { datePickerFormat: datePickerType }),
```

Comment:
One change I'd make that's unrelated to your work is making the `concept` property setter conditional:

```
          ...(selectedConcept &&  { concept: selectedConcept?.uuid }),
```

So that concept isn't set to an empty string (as is the case now) when `selectedConcept` is falsy.

Reply:
Updated the same condition for `EditQuestionModal`